### PR TITLE
lesson_6_mihaela_vaida

### PR DESCRIPTION
**Pull request by MihaelaGit**
Functions homework lesson 6 Mihaela Vaida

### CODE REVIEW

File: lesson_6_mihaela_vaida/functions_mihaela_vaida.py [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+for i in range(len(raw_data)):
+    coverages.append(raw_data[i][1])'''
+    
+def get_country(country_code):
+    country = \
+        'Albania' if country_code == 'AL' else \
+        'Romania' if country_code == 'RO' else \
+        'United Kingdom' if country_code == 'UK' \
+        else country_code
+    return country_code
```
Comment (mariuspodean) [inline lesson_6_mihaela_vaida/functions_mihaela_vaida.py]: Why go all the trouble of building **country** if you are returning country_code?

File: lesson_6_mihaela_vaida/functions_mihaela_vaida.py [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+'''coverages=[]
+for i in range(len(raw_data)):
+    coverages.append(raw_data[i][1])'''
+    
+def get_country(country_code):
+    country = \
+        'Albania' if country_code == 'AL' else \
+        'Romania' if country_code == 'RO' else \
+        'United Kingdom' if country_code == 'UK' \
+        else country_code
```
Comment (mariuspodean) [inline lesson_6_mihaela_vaida/functions_mihaela_vaida.py]: Getting the country based on a country code can be a lot simpler; basically you have here a mapping and you are fetching the country based on country code ;)

File: lesson_6_mihaela_vaida/functions_mihaela_vaida.py [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+               
+def prepare_dataset(desc, raw_data):
+    data = {}
+
+    country_year_data = []
+    for row in raw_data:        
+        country = get_country(row[0])
+        year_data = row[1]
+
+        country_year_data = [{'year': int(y), 'coverage': c} for y, c in zip(desc[1], year_data)]
```
Comment (mariuspodean) [inline lesson_6_mihaela_vaida/functions_mihaela_vaida.py]: The code it's a lot easier to read if you format it on multiple lines:
```python
country_year_data = [
    {
        'year': int(y),
        'coverage': c
    }
    for y, c in zip(desc[1], year_data)
]
```
Also, y and c can be replaced with more meaningful names

File: lesson_6_mihaela_vaida/functions_mihaela_vaida.py [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+                coverage = d['coverage']
+        result.append((get_country(country), coverage))
+
+    return {str(year): result}
+               
+
+def get_year_data_from_raw_dataset(raw_data, year, description=description):
+    year_idx = description[1].index(str(year))
+    print(f"year_idx: {year_idx}")
+    result = [(get_country(rd[0]), rd[1][year_idx]) for rd in raw_data]
```
Comment (mariuspodean) [inline lesson_6_mihaela_vaida/functions_mihaela_vaida.py]: formatting

```
result = [
    (get_country(rd[0]), rd[1][year_idx])
    for rd in raw_data
]
```

File: lesson_6_mihaela_vaida/functions_mihaela_vaida.py [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+
+    return {str(year): result}        
+
+    
+        
+
+def get_country_data(prepared_dataset, country):
+    country_data = [(x['year'], x['coverage']) for x in prepared_dataset[country]]
```
Comment (mariuspodean) [inline lesson_6_mihaela_vaida/functions_mihaela_vaida.py]: same as above for formatting

**Comment by Liitle** [comment]
legat de structura de foldere as pastra folderul `lesson_6` si as adauga in interior folderul `mihaela_vaida` cu fisierele de team

File: lesson_6_mihaela_vaida/functions_mihaela_vaida.py [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+('RO', ['47 ', '54 ', '58 ', '61 b', '68 ', '72 ', '76 ', '81 ', '84 ']),
+('RS', [': ', ': ', ': ', ': ', '64 ', ': ', '68 ', '73 ', '80 ']),
+('SE', ['91 ', '92 ', '93 ', '90 ', '91 ', '94 b', '95 ', '93 ', '96 ']),
+('SI', ['73 ', '74 ', '76 ', '77 ', '78 ', '78 ', '82 ', '87 ', '89 ']),
+('SK', ['71 ', '75 ', '78 ', '78 ', '79 ', '81 ', '81 ', '81 ', '82 ']),
+('TR', [': ', '47 ', '49 ', '60 ', '70 ', '76 ', '81 ', '84 ', '88 ']),
+('UK', ['83 ', '87 ', '88 ', '90 ', '91 ', '93 ', '94 ', '95 ', '96 ']),
+('XK', [': ', ': ', ': ', ': ', ': ', ': ', '89 ', '93 ', '93 '])
+]
+'''years=description[1]
```
Comment (Liitle) [inline lesson_6_mihaela_vaida/functions_mihaela_vaida.py]: This is a multiline string which is part of the code. If you want to comment code you can use `#` on each line or select the code and use `ctrl+/`(in pycharm) to auto comment/uncomment the code

File: lesson_6_mihaela_vaida/functions_mihaela_vaida.py [hunk 7]
```diff
@@ -0,0 +1,173 @@
...
+    countries.append(raw_data[i][0])'''
+
+
+'''coverages=[]
+for i in range(len(raw_data)):
+    coverages.append(raw_data[i][1])'''
+    
+def get_country(country_code):
+    country = \
+        'Albania' if country_code == 'AL' else \
```
Comment (Liitle) [inline lesson_6_mihaela_vaida/functions_mihaela_vaida.py]: it is a good approach but it only works for Albania Romania And United Kingdom. Can you do it for all the countries?

File: lesson_6_mihaela_vaida/functions_mihaela_vaida.py [hunk 8]
```diff
@@ -0,0 +1,173 @@
...
+        'Romania' if country_code == 'RO' else \
+        'United Kingdom' if country_code == 'UK' \
+        else country_code
+    return country_code               
+               
+def prepare_dataset(desc, raw_data):
```
Comment (Liitle) [inline lesson_6_mihaela_vaida/functions_mihaela_vaida.py]: `desc` can be renamed into `description`

File: lesson_6_mihaela_vaida/functions_mihaela_vaida.py [hunk 9]
```diff
@@ -0,0 +1,173 @@
...
+    data = {}
+
+    country_year_data = []
+    for row in raw_data:
```
Comment (Liitle) [inline lesson_6_mihaela_vaida/functions_mihaela_vaida.py]: you can include the variables in for 

`for country, year_data in raw_data`

File: lesson_6_mihaela_vaida/functions_mihaela_vaida.py (hunk 3 above)
Comment (Liitle) [inline lesson_6_mihaela_vaida/functions_mihaela_vaida.py]: nice

File: lesson_6_mihaela_vaida/functions_mihaela_vaida.py [hunk 10]
```diff
@@ -0,0 +1,173 @@
...
+        country_year_data = [{'year': int(y), 'coverage': c} for y, c in zip(desc[1], year_data)]
+        data[country] = country_year_data
+
+    return data               
+
+def get_year_data(prepared_dataset, year):
+    result = []
+    for country, data in prepared_dataset.items():
+        coverage = None        
+        for d in data:
```
Comment (Liitle) [inline lesson_6_mihaela_vaida/functions_mihaela_vaida.py]: there must be batter names for `d` and `data` 😃

**Comment by Liitle** [comment]
Nice job, keep up the good work

File: lesson_6_mihaela_vaida/functions_mihaela_vaida.py (hunk 10 above)
Comment (MihaelaGit) [reply lesson_6_mihaela_vaida/functions_mihaela_vaida.py]: I`ll try to keep that in mind next time, thank you :)

File: lesson_6_mihaela_vaida/functions_mihaela_vaida.py (hunk 7 above)
Comment (MihaelaGit) [reply lesson_6_mihaela_vaida/functions_mihaela_vaida.py]: Actually, it does not working for them either .. but  i was not able to solve that :(

**Comment by MihaelaGit** [comment]
Sincer, sunt de acord cu structura de folder dar momentan nu ma descurc sa le mut  acolo singura . Abia am reusit sa le pun pe git, dupa cateva erori si m-a ajutat si un coleg de   la serviciu . Dar o sa incerc mai bine  data urmatoare

File: lesson_6_mihaela_vaida/functions_mihaela_vaida.py (hunk 6 above)
Comment (MihaelaGit) [reply lesson_6_mihaela_vaida/functions_mihaela_vaida.py]: done

File: lesson_6_mihaela_vaida/functions_mihaela_vaida.py (hunk 8 above)
Comment (MihaelaGit) [reply lesson_6_mihaela_vaida/functions_mihaela_vaida.py]: renamed

File: lesson_6_mihaela_vaida/functions_mihaela_vaida.py (hunk 5 above)
Comment (MihaelaGit) [reply lesson_6_mihaela_vaida/functions_mihaela_vaida.py]: done

File: lesson_6_mihaela_vaida/functions_mihaela_vaida.py (hunk 4 above)
Comment (MihaelaGit) [reply lesson_6_mihaela_vaida/functions_mihaela_vaida.py]: done

File: lesson_6_mihaela_vaida/functions_mihaela_vaida.py (hunk 3 above)
Comment (MihaelaGit) [reply lesson_6_mihaela_vaida/functions_mihaela_vaida.py]: renamed

File: lesson_6_mihaela_vaida/functions_mihaela_vaida.py (hunk 2 above)
Comment (MihaelaGit) [reply lesson_6_mihaela_vaida/functions_mihaela_vaida.py]: removed Country

**Review by mariuspodean — APPROVED** [review]

**Pull request merged**